snoop/management/commands/resetindex.py

```diff
@@ -43,9 +44,20 @@
 class Command(BaseCommand):
     help = "Reset the elasticsearch index"
 
-    def handle(self, **options):
-        es.indices.delete(settings.SNOOP_ELASTICSEARCH_INDEX, ignore=[400, 404])
-        es.indices.create(settings.SNOOP_ELASTICSEARCH_INDEX, {
+    def add_arguments(self, parser):
+        parser.add_argument(
+            'collection_slug',
+            help='Slug of the collection that needs its index reset.'
+        )
+
+    def handle(self, collection_slug, **options):
+        try:
+            collection = models.Collection.objects.get(slug=collection_slug)
+        except models.Collection.DoesNotExist:
```

Comment:
Eh, you can let the exception be raised, the message will be descriptive enough. If you do want to hande it, make sure the program exits with an error code, `sys.exit(1)` maybe.